src/main/java/com/bae/service/RecipeService.java

```diff
@@ -1,5 +1,74 @@
 package com.bae.service;
 
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.bae.exceptions.RecipeNotFoundException;
+import com.bae.persistence.domain.Recipe;
+import com.bae.persistence.repository.RecipeRepository;
+
+@Service
 public class RecipeService {
 
-}
+	private RecipeRepository repository;
+	
+	@Autowired
+	public RecipeService(RecipeRepository repository) {
+		this.repository = repository;
+	}
+
+	public List<Recipe> findAllRecipe() {
+		return repository.findAll();
+	}
+
+	public Recipe createRecipe(Recipe recipe) {
+		if (!recipe.getRecipeTitle().matches("[a-zA-Z]+{5,30}")) {
+			throw new IllegalStateException("Invalid Recipe Name. Please enter a recipe name between 5 and 30 letters from A to Z.");
+		}
+		return this.repository.save(recipe);
+		
+
+		if (!recipe.getRecipeMethod().matches("^\\W*(?:\\w+\\b\\W*){100,600}$")) {
+			throw new IllegalStateException();
+		}
+		return this.repository.save(recipe);
+		
+		if (!recipe.getCookTime().matches("^(1[0-2]|0?[1-9]):([0-5]?[0-9])$")) {

```

Comment:
The error here is because regex .matches treats the expression you have defined with the funky set of symbols as a string, so the thing it is checking must also be a string
```suggestion
		else if (!Double.toString(recipe.getCookTime()).matches("^(1[0-2]|0?[1-9]):([0-5]?[0-9])$")) {
```